### PR TITLE
docs(users): Add CoreWeave and ConciergeRender

### PR DESCRIPTION
**Pull request by salanki**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [X] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlike to be merged.
* [X] Optional. I've added My organization is added to the USERS.md.
* [X] I've signed the CLA and required builds are green. 

Adds us to the users list. CoreWeave is a GPU-specialized IaaS, taking on AWS and GCP with a leaner approach to resource allocation and lower end-user cost. We use argo to execute and orchestrate workflows in parallel over 6,000 GPUs per Kubernetes cluster for our own end-user products (CGI rendering) and for our Cloud clients (in many industries).

### CODE REVIEW

File: USERS.md [hunk 1]
```diff
@@ -63,3 +63,5 @@ Currently, the following organizations are **officially** using Argo Workflows:
 1. [Threekit](https://www.threekit.com/)
 1. [Tiger Analytics](https://www.tigeranalytics.com/)
 1. [Wavefront](https://www.wavefront.com/)
+1. [CoreWeave Cloud](https://www.coreweave.com)
```
Comment (alexec) [inline USERS.md:66]: please sort alphabetically!

Comment (salanki) [reply USERS.md]: Done

**Comment by salanki** [comment]
Rebased on top of master, will be up to date for a couple of minutes I hope!

**Comment by alexec** [comment]
thank you!

**Pull request merged**